src/comp/Header.tsx

```diff
@@ -21,36 +30,83 @@ class Header extends React.Component<HeaderProps, HeaderState> {
     });
   }
 
+  handleResize = () => {
+    if (window.innerWidth <= 970) {
+      this.setState({ showMenuIcon: true });
+    } else {
+      this.setState({
+        showMenuIcon: false,
+        showMenuList: false,
+      });
+    }
+    if (window.innerWidth <= 544) {
+      this.setState({ moveSearchContents: true });
+    } else {
+      this.setState({ moveSearchContents: false });
+    }
+  };
+
+  changeMenuListStatus = (flag: boolean) => {
+    this.setState({ showMenuList: flag });
+  };
+
+  menuButtonIcon = (
+    <IconButton
+      onClick={() => this.changeMenuListStatus(!this.state.showMenuList)}
+    >
+      <Menu className={styles.menuIcon} fontSize="large" />
+    </IconButton>
+  );
+
   render() {
+    let menuContents: JSX.Element, searchContent: JSX.Element;
+    const menuButtonList = (
+      <HeaderMenu
+        showMenuIcon={this.state.showMenuIcon}
+        moveSearchContents={this.state.moveSearchContents}
+        handleSearchBoxChange={this.handleSearchBoxChange}
```

Comment:
```suggestion
        handleSearchBoxChange={this.handleSearchBoxChange.bind(this)}
```
検索ボックスの値を変更したときにエラーがでてつらいです